lib/cisco_node_utils/command_reference.rb

```diff
@@ -130,6 +109,43 @@ def define_getter(key, value)
       end
     end
 
+    # curried function to define a getter method body that performs key-value
+    # substitution
+    def key_substitutor(config_key, value)
+      lambda do |**args|
+        result = []
+        value.each do |line|
+          replace = line.scan(/<(\S+)>/).flatten.map(&:to_sym)
+          replace.each do |item|
+            line = line.sub("<#{item}>", args[item].to_s) if args.key?(item)
+          end
+          result.push(line) unless /<\S+>/.match(line)
+        end
+        if result.empty?
+          fail ArgumentError,
+               "Arguments given to #{config_key} yield empty result"
+        end
+        preprocess_value(result)
+      end
+    end
+
+    # curried function to define a getter method body that performs
+    # printf-style substitution
+    def printf_substitutor(config_key, value)
+      arg_c = value.join.scan(/%/).length
+      lambda do |*args|
+        unless args.length == arg_count
```

Comment:
good catch; fixed in 96eaaea
(This is what happens when line-length requirements lead me to rename variables to shorter names.)